platform/viewer/cypress/integration/common/OHIFCornerstoneToolbar.spec.js

```diff
@@ -0,0 +1,509 @@
+            .should('have.class', 'active')
+        })
+      
+      //Get image position from cornerstone and check if y axis was modified
+      let cornerstone;
+      let currentPan;
+
+      cy.window()
+        .its('cornerstone')
+        .then((c) => {
+          cornerstone = c;
+          currentPan = () => cornerstone.getEnabledElements()[0].viewport.translation;
+        });
+
+      cy.get('.viewport-element')
+        .trigger('mousedown', 'center', { which: 1 })
+        .trigger('mousemove', 'bottom', { which: 1 })
+        .trigger('mouseup', 'bottom')
+        .then(() => {
+          expect(currentPan().y > 0).to.eq(true);
+        });
+
+        cy.resetViewport();
+    });
+
+
+    it('checks if Length annotation can be added on viewport and on measurements panel', () => {   
+      //Click on button and vefiry if icon is active on toolbar
+      cy.get('.toolbar-button:nth-child(6)').as('lengthBtn')
+        .click()
+        .then(($lengthbtn) =>{
+          cy.wrap($lengthbtn)
+            .should('have.class', 'active')
+        })
+
+      //Add annotation on the viewport
+      const initPos = [150, 150];
+      const finalPos = [150, 170];
+      cy.addLine('.cornerstone-canvas', initPos, finalPos)
+      
+      //Verify if measurement annotation was added into the measurements panel
+      cy.get('.pull-right > .RoundedButtonGroup > .roundedButtonWrapper > .roundedButton')
+        .as('measurementsBtn')
+        .click()
+        .then($measurementsBtn => {
+          cy.get('section.sidepanel.from-right')
+            .as('measurementsPanel')
+            .should('be.visible');
+            
+          cy.get('.measurementItem')
+            .its('length')
+            .should('be.eq', 1);
+
+          cy.wrap($measurementsBtn)
+          .click();
+        })
+    });
+
+
+    it('checks if Angle annotation can be added on viewport and on measurements panel', () => {   
+      //Click on button and vefiry if icon is active on toolbar
+      cy.get('.toolbar-button:nth-child(8)').as('angleBtn')
+        .click()
+        .then(($angleBtn) =>{
+          cy.wrap($angleBtn)
+            .should('have.class', 'active')
+        })
+
+      //Add annotation on the viewport
+      const initPos = [180, 390];
+      const midPos = [300, 410];
+      const finalPos = [180, 450];
+      cy.addAngle('.cornerstone-canvas', initPos, midPos, finalPos);
+
+      //Verify if measurement annotation was added into the measurements panel
+      cy.get('.pull-right > .RoundedButtonGroup > .roundedButtonWrapper > .roundedButton')
+        .as('measurementsBtn')
+        .click()
+        .then($measurementsBtn => {
+          cy.get('section.sidepanel.from-right')
+            .as('measurementsPanel')
+            .should('be.visible');
+            
+          cy.get('.measurementItem')
+            .its('length')
+            .should('be.eq', 2);
+
+          cy.wrap($measurementsBtn)
+          .click();
+        })
```

Comment:
This is a really good assertion to test ^_^
I think if we apply the lessons learned in other tests, we can clean it up to read a little easier.